scripts/sanitycheck

```diff
@@ -1448,6 +1454,19 @@ class MakeGenerator:
                                 goal.fail(thread_status)
                         else:
                             goal.success()
+                        if keep_files is not None:
+                            (o, keep_paths) = keep_files[name]
+                            for dirpath, dirnames, filenames in os.walk(o, topdown=False):
+                                for name in filenames:
+                                    path = os.path.join(dirpath, name)
+                                    if path not in keep_paths:
+                                        os.remove(path)
+                                for name in dirnames:
```

Comment:
Can add comment to make this quicker to understand:

```

# Remove empty directories and symbolic links to directories
for dirname in dirnames:
    ...
```

Reply:
Done